plugins/modules/firewall_alias.py

```diff
@@ -0,0 +1,273 @@
+        and _"
+    type: str
+    required: true
+  type:
+    description:
+      - The type used for the Alias
+      - hosts (Single hosts by IP or Fully Qualified Domain Name or host
+        exclusions (starts with '!' sign))
+      - networks (Entire network p.e. 192.168.1.1/24 or network exclusion eg
+        !192.168.1.0/24)
+      - ports (Port numbers or a port range like 20:30)
+      - urls (A table of IP addresses that are fetched once)
+      - urltable (A table of IP addresses that are fetched on regular
+        intervals.)
+      - geoip (Select countries or whole regions) disclaimer -> validation is not supported at this point
+      - networkgroup (Combine different network type aliases into one)
+      - macaddress (MAC address or partial mac addresses like f4:90:ea)
+      - bgpasn (Maps autonomous system (AS) numbers to networks where they
+        are responsible for) supported >= version 23.7
+      - dynamicipv6host (A Host entry that will auto update on a
+        prefixchange) supported >= version 23.7
+      - opnvpngroup (Map user groups to logged in OpenVPN users) supported
+        >= version 23.1
+      - internal (Internal aliases which are managed by the product)
+      - external (Externally managed alias, this only handles the
+        placeholder. Content is set from another source (plugin, api call,
+        etc))
+    type: str
+    choices:
+      - host
+      - network
+      - port
+      - url
+      - urltable
+      - geoip
+      - networkgroup
+      - macaddress
+      - bgpasn
+      - dynamicipv6host
+      - opnvpngroup
+      - internal
+      - external
+    required: true
+  content:
+    description:
+      - Content of the alias
+    type: list
+    elements: str
+    required: false
+  statistics:
+    description:
+      - Maintain a set of counters for each table entry
+    type: bool
+    required: false
+    default: false
+  description:
+    description:
+      - Description of the Alias
+    type: str
+    required: false
+  refreshfrequency:
+    description:
+      - The frequency that the list will be refreshed, in days + hours, so 1 day
+        and 8 hours means the alias will be refreshed after 32 hours.
+    type: str
+    required: false
+  interface:
+    description:
+      - Select the interface for the V6 dynamic IP
+    type: str
+    required: false
+  state:
+    description: Weather alias should be added or removed.
+    required: false
+    type: str
+    default: present
+    choices: [present, absent]
+'''
+
+EXAMPLES = r'''
+- name: Create an Host Alias with the content 10.0.0.1
+  puzzle.opnsense.firewall_alias:
+    name: TestAliasTypeHost
+    type: host
+    statistics: false
+    description: Test Alias with type Host
+    content: 10.0.0.1
+
+- name: Create a URL Alias with the content www.puzzle.ch
+  puzzle.opnsense.firewall_alias:
+    name: TestAliasTypeURL
+    type: urls
```

Comment:
```suggestion
    type: url
```

Reply:
Thanks for the suggestion! Fixed in f139dc0.